### PR TITLE
Crass, nvamelichev merge

### PR DESCRIPTION
**Pull request by Evidlo**
Merging contributions from @crass, and @nvamelichev.

Summary of changes

- unicode fixes
- python3 support
- twofish encryption support for KDB3 databases (see [here](http://keepass.info/help/base/security.html))
- lots and lots of small fixes and code cleanup

@pschmitt, @phpwutz Would you care to take a quick look and see if everything checks out?  The tests succeeded for both python2 and python3 and I also successfully ran the tests for pykeepass.

### CODE REVIEW

**Comment by pschmitt** [comment]
Nice! This should be worth a new release.

The only thing missing here is an up to date `requirements.txt` file.

File: setup.py [hunk 1]
```diff
@@ -1,15 +1,28 @@
...
+        "pycrypto>=2.6.1",
+        "colorama>=0.3.2"
+    ],
+    classifiers=[
+        "Development Status :: 3 - Alpha",
+        "Programming Language :: Python :: 2.7",
+        "Programming Language :: Python :: 3.2",
+        "Programming Language :: Python :: 3.3",
+    ],
     test_suite="tests"
```
Comment (pschmitt) [inline setup.py]: Lines 27 and 14 are the same. This leads to the following error when trying to install using pip:
```Processing /home/pschmitt/dev/libkeepass
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-fOzaVK-build/setup.py", line 27
        test_suite="tests"
    SyntaxError: keyword argument repeated

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-fOzaVK-build/
```

**Review by pschmitt — CHANGES_REQUESTED** [review]

**Review by pschmitt — APPROVED** [review]

**Comment by pschmitt** [comment]
@phpwutz any objections against merging this?

**Comment by phpwutz** [comment]
from a superficial inspection, no objections 👍

**Pull request merged**